File: app/src/main/java/com/bapplications/maplemobile/views/GameViewController.java
package com.bapplications.maplemobile.views;

import android.annotation.SuppressLint;
import android.util.Log;

import android.view.MotionEvent;
import android.view.View;

public class GameViewController {

    private KeyAction key;
    private boolean isPressed = false;

    @SuppressLint("ClickableViewAccessibility")
    public GameViewController(KeyAction key, View view, UIControllers uiControllers) {
        this.key = key;
        switch (key.getType()) {
            case CONTINUES_CLICK:
                view.setOnTouchListener((view1, motionEvent) -> {
                    if (motionEvent.getAction() == MotionEvent.ACTION_DOWN) {
                        isPressed = true;
                    } else if (motionEvent.getAction() == MotionEvent.ACTION_UP) {
                        isPressed = false;
                    }
                    return true;
                });
                break;
            case SINGLE_CLICK:
                view.setOnClickListener(view12 -> uiControllers.onClick(key));
                break;
        }
    }

    public boolean isPressed() {
        return isPressed;
    }
}

File: app/src/main/java/com/bapplications/maplemobile/ui/GameViewController.kt
package com.bapplications.maplemobile.ui

import android.view.MotionEvent
import android.view.View
import com.bapplications.maplemobile.input.InputAction

class GameViewController(button: View, val type: InputAction.Type) {

    var isPressed = false
    var isReleased = false
    get() {
        val res = field
        field = false
        return res
    }

    init {
        button.setOnTouchListener { _: View?, motionEvent: MotionEvent ->
            if (!isPressed && motionEvent.action == MotionEvent.ACTION_DOWN) {
                isPressed = true
            } else if (motionEvent.action == MotionEvent.ACTION_UP) {
                isPressed = false
                isReleased = true
            } else if( type == InputAction.Type.SINGLE_CLICK) {
                isPressed = false;
            }
            true
        }
    }
}